PennMobile/Weekly Schedule/WeeklyScheduleModelManager.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  ScheduleNetworkManager.swift
+//  PennMobile
+//
+//  Created by CHOI Jongmin on 16/2/2020.
+//  Copyright © 2020 PennLabs. All rights reserved.
+//
+
+import Foundation
+import SwiftyJSON
+
+final class WeeklyScheduleModelManager {
+        
+    static let instance = WeeklyScheduleModelManager()
+    private init() {}
+    
+    func fetchModel(_ completion: @escaping (_ model: WeeklyScheduleTableViewModel?) -> Void) {
+        var model: WeeklyScheduleTableViewModel? = WeeklyScheduleTableViewModel()
+        
+        if let courses = UserDefaults.standard.getCourses() {
+            if courses.isEmpty {
+                
+            }
+            model = try? WeeklyScheduleTableViewModel(courses: courses)
+            completion(model)
+        } else {
+            completion(nil)
+        }
+    }
+}
```

Comment:
This isn't really necessary because the data is stored locally on the phone. A callback is only needed if we're making an asynchronous request (i.e. network request), which we're not in this case.

Reply:
I removed the callback and instead segmented the initialization of the model to an external file called **CourseScheduleTableViewModel** in the newest commit. Is this ideal or would you prefer it if all of it were in a single file?

Reply:
Yes, that works, although it doesn't make much sense to put fetchModel in a callback either because it's not asynchronous. Let's just return the model directly or put this in viewDidLoad.